state/merkle.go

```diff
@@ -61,16 +53,18 @@ func (s *State) Commit(version uint64) ([]byte, error) {
 		return nil, err
 	}
 
+	// store a new version
 	var hash []byte
-	if s.persistent {
-		if s.committed.Tree.Size() > 0 || s.committed.Tree.LatestVersion() > 0 {
-			hash, err = s.committed.Tree.SaveVersion(version)
-			if err != nil {
-				return nil, err
-			}
+	if s.committed.Tree.Size() > 0 || s.committed.Tree.LatestVersion() > 0 {
+		hash, err = s.committed.Tree.SaveVersion(version)
+		if err != nil {
+			return nil, err
 		}
-	} else {
-		hash = s.committed.Tree.Hash()
+	}
+
+	// release an old version
+	if version > s.historySize {
+		s.committed.Tree.DeleteVersion(version - s.historySize)
```

Comment:
Nice poor-man's pruning!

Reply:
Yup, hardcoded length, enough to get it out. We can work on other techniques later, but refactoring will change a lot, so it can wait.